### PR TITLE
Add struct formatting to sway-fmt-v2

### LINKED ISSUE #2053: `sway-fmt-v2` should be able to format `structs`
This issue is for tracking the progress of `sway-fmt-v2` implementation for structs.

related & part of https://github.com/FuelLabs/sway/issues/1516.

### PR DESCRIPTION
**Pull request by kayagokalp**
closes #2053.

~[WIP] I am still testing this, opening up for visibility.~

- [x] Implement alignment
- [x] Add test for alignment

### CODE REVIEW

**Review by mitchmindtree — COMMENTED** [review]
@kayagokalp @eureka-cpu you guys are killing it!

File: sway-fmt-v2/src/fmt.rs
Comment (eureka-cpu) [inline sway-fmt-v2/src/fmt.rs:142]: This will need to be 4 spaces, since the default is 4

File: sway-fmt-v2/src/items/item_struct.rs
Comment (eureka-cpu) [inline sway-fmt-v2/src/items/item_struct.rs:180]: Add a variable, ` let extra_width = formatter.config.whitespace.tab_spaces`

File: sway-fmt-v2/src/items/item_struct.rs
Comment (eureka-cpu) [inline sway-fmt-v2/src/items/item_struct.rs:186]: And use it here

File: sway-fmt-v2/src/items/item_struct.rs
Comment (eureka-cpu) [inline sway-fmt-v2/src/items/item_struct.rs:191]: here

File: sway-fmt-v2/src/items/item_struct.rs
Comment (eureka-cpu) [inline sway-fmt-v2/src/items/item_struct.rs:201]: same here, except no need to store in a variable just call it directly

File: sway-fmt-v2/src/fmt.rs
Comment (eureka-cpu) [inline sway-fmt-v2/src/fmt.rs:105]: instead of using `get_formatter()` you can do `Formatter::default()`

**Review by eureka-cpu — CHANGES_REQUESTED** [review]

File: sway-fmt-v2/src/items/item_struct.rs
Comment (mitchmindtree) [inline sway-fmt-v2/src/items/item_struct.rs:57]: When writing doc comments, try to be considerate of the markdown formatting, even if it's just for private functions. I believe they can show up in an LSP-enabled editor with markdown formatting when hovered, and they can also be generated using `cargo doc --document-private-items` which I personally find handy when familiarising with a new code-base.

Here's how the current formatting looks:

Screenshot from 2022-06-23 11-22-53

Here's how it can look with some slight reformatting:

Screenshot from 2022-06-23 11-27-27

```rust
/// Format the struct if the multiline is passed as false struct will be formatted into a single line.
///
/// ## Examples
///
/// (multiline : false):
///
/// ```rust,ignore
/// struct Foo { bar: u64,  baz: bool }
/// ```
///
/// (multiline : true):
///
/// ```rust,ignore
/// struct Foo {
///     bar: u64,
///     baz: bool,
/// }
/// ```
```

**Review by mitchmindtree — APPROVED** [review]
Looking good. Thanks for the extensive tests, they make it much easier to review!

I've left a comment around good doc comment practise. It shouldn't block these changes, and I realise there are already a lot of places throughout the repo where formatting is a bit broken, but it would be sweet if we could try to be more aware of this going forward.

File: sway-fmt-v2/src/items/item_struct.rs
Comment (kayagokalp) [reply sway-fmt-v2/src/items/item_struct.rs:57]: Oh, that is really handy thanks. I will be using that to ensure that docs seem as I intended 👍

Comment (eureka-cpu) [reply sway-fmt-v2/src/items/item_struct.rs:57]: @kayagokalp did you want to add this, or should I go ahead and approve as is?

Comment (kayagokalp) [reply sway-fmt-v2/src/items/item_struct.rs:57]: @eureka-cpu After merging this I will be completing https://github.com/FuelLabs/sway/pull/2097. So I was planning to fix the doc there so that mitch won't need to approve again but if you prefer the other way we can go that way too.

**Review by eureka-cpu — APPROVED** [review]

**Pull request merged**
